src/Transform/ONNX/Combine.td

```diff
@@ -75,6 +75,12 @@ def CastEliminationPattern : Pat<
 	(replaceWithValue $arg),
   [(HasSameElementType $arg, $type)]>;
 
+// (onnx.Cast (onnx.Cast(%X, $type2), $type1)) = (onnx.Cast(%X, $type1))
+def CastPropagationPattern : Pat<
```

Comment:
As we talked about, this affects result if we optimize away lower precision type.
For example, if we had `cast<fp32>(cast<int8>(x))`, this would not produce same result as `cast<fp32>(x)`.
This might be okay if we were doing static quantization, since this might improve accuracy.
However, if the model was trained using quantization aware training, the training model would have taken clip to [-128, 127] of cast<int8>(x) into account in its parameters, and optimizing away that operation may produce lower accuracy results.

Reply:
Thanks for pointing this out. I'm moving this into our codebase and creating a follow up item to investigate if this makes our true accuracy better or worse.